rpmlint/permissions.py

```diff
@@ -0,0 +1,202 @@
+# vim: sw=4 et sts=4 ts=4 :
+#############################################################################
+# Author        : Matthias Gerstner
+# Purpose       : reusable code for parsing permissions/chkstat profiles
+#############################################################################
+
+import copy
+import os
+
+
+class PermissionsEntry:
+
+    # source profile path
+    profile = None
+    # source profile line nr
+    linenr = None
+    # target path
+    path = None
+    owner = None
+    group = None
+    # mode as integer
+    mode = None
+    caps = []
+    # related paths from variable expansions
+    related_paths = []
+
+    def __init__(self, _profile, _line_nr):
+
+        self.profile = _profile
+        self.linenr = _line_nr
+
+    def __str__(self):
+
+        ret = '{}:{}: {path} {owner}:{group} {mode}'.format(
```

Comment:
Please use rather `f-strings`, it's nicely explained here:
https://realpython.com/python-f-strings/